tests/unit/lms/resources/_js_config/__init___test.py

```diff
@@ -153,14 +157,89 @@ def test_it(
         }
 
+    @pytest.mark.usefixtures("grouping_plugin")
+    def test_configures_the_client_with_course_group(
+        self, js_config, grouping_service, course, assignment
+    ):
+        grouping_service.get_launch_grouping_type.return_value = Grouping.Type.COURSE
+
+        js_config.enable_lti_launch_mode(course, assignment)
+        config = js_config.asdict()
+
+        assert config["hypothesisClient"]["services"][0]["groups"] == [
+            Any.string.matching("^group:.*@lms.hypothes.is")
+        ]
+        assert not config["api"]["sync"]
+
+    @pytest.mark.usefixtures("grouping_plugin")
+    @pytest.mark.parametrize(
+        "grouping_type", [Grouping.Type.SECTION, Grouping.Type.GROUP]
+    )
+    def test_configures_the_client_to_fetch_the_groups_over_RPC(
+        self,
+        js_config,
+        grouping_service,
+        grouping_type,
+        course,
+        assignment,
+        grouping_plugin,
+        pyramid_request,
+    ):
+        grouping_service.get_launch_grouping_type.return_value = grouping_type
+        pyramid_request.lti_params["context_id"] = "CONTEXT_ID"
+        pyramid_request.params["learner_canvas_user_id"] = "CANVAS_USER_ID"
+        grouping_service.get_launch_grouping_type.return_value = grouping_type
+        grouping_plugin.get_group_set_id.return_value = "GROUP_SET_ID"
+
+        js_config.enable_lti_launch_mode(course, assignment)
+        config = js_config.asdict()
+
+        assert (
+            config["hypothesisClient"]["services"][0]["groups"] == "$rpc:requestGroups"
+        )
+        sync_config = config["api"]["sync"]
+        assert sync_config == {
+            "authUrl": "http://example.com/welcome",
+            "path": "/api/sync",
+            "data": {
+                "resource_link_id": assignment.resource_link_id,
+                "lms": {"product": pyramid_request.product.family},
+                "context_id": "CONTEXT_ID",
+                "group_set_id": "GROUP_SET_ID",
+                "group_info": {
+                    "context_id": "CONTEXT_ID",
+                    "custom_canvas_course_id": "test_course_id",
+                },
+                # This is only actually true for Canvas, but we do it for all
+                # LMS products at the moment
+                "gradingStudentId": "CANVAS_USER_ID",
+            },
+        }
+
+        # Confirm we pass the schema for the sync end-point
+        APISyncSchema(pyramid_request).load(sync_config["data"])
+
+    @pytest.mark.usefixtures("with_provisioning_disabled")
+    def test_client_config_is_empty_if_provisioning_feature_is_disabled(
+        self, js_config
+    ):
+        js_config.enable_lti_launch_mode(course, assignment)
+        config = js_config.asdict()["hypothesisClient"]
+
+        assert config == {}
+
+    @pytest.fixture
+    def with_provisioning_disabled(self, context):
+        context.application_instance.provisioning = False
```

Comment:
I don't actually know what `provisioning` is in LMS.

Reply:
I'm not clear. I think it's the automatic creation of users in H? I have no idea if it works without it any more. We don't test it I don't think.

Maybe this is worth a chat to see if this is a legacy thing that could be stripped out?

A quick look at the numbers seems to suggest this is still in play, but not a lot recently:

 * 444 total with provisioning disabled
 * 201 which we have last launched dates for
 * 4 launched this year